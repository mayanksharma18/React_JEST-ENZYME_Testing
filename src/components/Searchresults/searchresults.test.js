import React from "react"
import Enzyme ,{shallow} from 'enzyme'
import SearchResults from './index';
import Adapter from 'enzyme-adapter-react-16'
import { exportAllDeclaration } from "@babel/types";

Enzyme.configure({adapter: new Adapter()})


const mockArticle =[{
    webUrl:"http://google.com",
    webTitle:"This is google"
    
}]

describe("Searchresults component",()=>{
    test("renders",()=>{
        const wrapper = shallow(<SearchResults article={mockArticle}/>)
        expect(wrapper).toMatchSnapshot();
    })
    test("when articles is empty",()=>{
        const wrapper=shallow(<SearchResults />);
        expect(wrapper).toMatchSnapshot()
    })

    test("doesnt break without articles",()=>{
        const wrapper=shallow(<SearchResults/>)
        expect(wrapper.find("li")).toHaveLength(0)
    })
    
    test("doesnt break with an empty array",()=>{
        const wrapper=shallow(<SearchResults articles={[]} />)

        expect(wrapper.find("li")).toHaveLength(0)
    })

})