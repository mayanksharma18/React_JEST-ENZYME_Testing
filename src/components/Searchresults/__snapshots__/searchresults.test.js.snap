// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchresults component renders 1`] = `ShallowWrapper {}`;

exports[`Searchresults component when articles is empty 1`] = `ShallowWrapper {}`;
